asu/build.py

```diff
@@ -29,187 +22,90 @@ def build(req: dict):
-            report_error("Bad Checksum")
+        return returncode, stdout, stderr
 
-        (cache_workdir).mkdir(parents=True, exist_ok=True)
-
-        job.meta["imagebuilder_status"] = "unpack_imagebuilder"
+    def report_error(msg):
+        log.warning(f"Error: {msg}")
+        job.meta["detail"] = f"Error: {msg}"
         job.save_meta()
+        raise
 
-        extract_archive = subprocess.run(
-            ["tar", "--strip-components=1", "-xf", ib_archive, "-C", subtarget],
-            cwd=cache / target,
-        )
-
-        if extract_archive.returncode:
-            report_error("Failed to unpack ImageBuilder archive")
-
-        log.debug(f"Extracted TAR {ib_archive}")
-
-        (cache / target / ib_archive).unlink()
-
-        for key in req["branch_data"].get("extra_keys", []):
-            fingerprint = fingerprint_pubkey_usign(key)
-            (cache_workdir / "keys" / fingerprint).write_text(
-                f"untrusted comment: ASU extra key {fingerprint}\n{key}"
-            )
-
-        repos_path = cache_workdir / "repositories.conf"
-        repos = repos_path.read_text()
-
-        extra_repos = req["branch_data"].get("extra_repos")
-        if extra_repos:
-            log.debug("Found extra repos")
-            for name, repo in extra_repos.items():
-                repos += f"\nsrc/gz {name} {repo}"
-
-        repos_path.write_text(repos)
-        log.debug(f"Repos:\n{repos}")
-
-        if (Path.cwd() / "seckey").exists():
-            # link key-build to imagebuilder
-            (cache_workdir / "key-build").symlink_to(Path.cwd() / "seckey")
-        if (Path.cwd() / "pubkey").exists():
-            # link key-build.pub to imagebuilder
-            (cache_workdir / "key-build.pub").symlink_to(Path.cwd() / "pubkey")
-        if (Path.cwd() / "newcert").exists():
-            # link key-build.ucert to imagebuilder
-            (cache_workdir / "key-build.ucert").symlink_to(Path.cwd() / "newcert")
+    req["store_path"].mkdir(parents=True, exist_ok=True)
 
-    def download_file(filename: str, dest: str = None):
-        """Download file from upstream target path
+    job = job or get_current_job()
+    job.meta["detail"] = "init"
+    job.save_meta()
 
-        The URL points automatically to the targets folder upstream
+    log.debug(f"Building {req}")
 
-        Args:
-            filename (str): File in upstream target folder
-            dest (str): Optional path to store the file, default to target
-                        cache folder
-        """
-        log.debug(f"Downloading {filename}")
-        r = requests.get(
-            req["upstream_url"]
-            + "/"
-            + req["branch_data"]["path"].format(version=req["version"])
-            + "/targets/"
-            + req["target"]
-            + "/"
-            + filename
-        )
+    podman = PodmanClient(
+        base_url="ssh://user@localhost:50622/tmp/podman.sock",
```

Comment:
SSH to localhost looks strange to me. Isn't there a URI format which would just use the local socket (ie. something like `file:///tmp/podman.sock`)?

Reply:
This is a bit of an awkward thing when running Bodman on MacOS, it's running in its own VM to have a Linux Kernel running. The official Podman is incredibly slow and using alternatives are focused on Docker. Long story short I'll make this user configurable.